portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,40 +15,58 @@
 package com.google.sps.servlets;
 
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 
 import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> quotes;
+  private List<Comment> comments = new ArrayList<>();
 
   @Override
-  public void init() {
-    quotes = new ArrayList<>();
-    quotes.add(
-        "A ship in port is safe, but that is not what ships are for.");
-    quotes.add("They told me computers could only do arithmetic. - Grace Hopper");
-    quotes.add("A ship in port is safe, but that's not what ships are built for. - Grace Hopper");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String json = new Gson().toJson(comments);
+    response.getWriter().println(json);
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJsonUsingGson(quotes));
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form
+    String username = getParameter(request, "user-name", "").trim();
```

Comment:
A comment on the logic flow here:
Since the default value for `username` and `commentText` are empty, you probably want to treat them as invalid input instead of storing empty ones in the DB.

In that case, consider making `getParameter` return Optional<String> instead.